src/utils/scripts.ts

```diff
@@ -1,72 +1,108 @@
-import { spawn } from "child_process";
+/* eslint-disable no-console */
+import { execSync, spawn } from "child_process";
 import { UserError } from "../errors.js";
 import colors from "colors";
 import _ from "lodash";
 import { Logger } from "tslog";
-import { execaCommand } from "execa";
-import { debugLogger, log } from "./logging.js";
+import { FunctionConfiguration } from "../models/projectConfiguration.js";
+import { YamlProjectConfiguration } from "../projectConfiguration/yaml/v2.js";
+import { FunctionType } from "../projectConfiguration/yaml/models.js";
+import getProjectInfoByName from "../requests/getProjectInfoByName.js";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isFunctionConfiguration(value: any): value is FunctionConfiguration {
+    return (
+        value instanceof FunctionConfiguration ||
+        (value &&
+            typeof value === "object" &&
+            typeof value.name === "string" &&
+            typeof value.path === "string" &&
+            typeof value.handler === "string" &&
+            typeof value.entry === "string" &&
+            Object.values(FunctionType).includes(value.type))
+    );
+}
 
-export async function expandFunctionURLVariablesFromScripts(
-    scripts: string | string[] | undefined,
-    functions:
-        | {
-              name: string;
-              url: string;
-          }[]
-        | undefined,
-): Promise<string | string[] | undefined> {
-    if (!scripts) {
-        return;
-    }
+export async function evaluateVariable(
+    configuration: YamlProjectConfiguration,
+    stage: string,
+    path: string,
+    field: string,
+): Promise<string> {
+    const keys = path.split(".");
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let value: any = configuration;
+
+    for (let i = 0; i < keys.length; i++) {
+        const key = keys[i];
+
+        if (Array.isArray(value)) {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            value = value.find((item: any) => item.name === key);
```

Comment:
Same here, maybe something like this:
```suggestion
            value = value.find((item: { name: string | undefined }) => item.name === key);
```